src/Confluent.Kafka/Producer.cs

```diff
@@ -315,6 +322,22 @@ public Producer(IEnumerable<KeyValuePair<string, object>> config, bool manualPol
                 callbackCts = new CancellationTokenSource();
                 callbackTask = StartPollTask(callbackCts.Token);
             }
+
+            // note: ConcurrentDictionary.GetorAdd() method is not atomic
+            this.topicHandlerFactory = (string topicName) =>
+            {
+                if (topicHandles.ContainsKey(topicName))
```

Comment:
This additional check is unnecessary. factory handlers will be for a topic handle creation always.
The rest merge looks good.